viewer/core/src/public/migration/Cognite3DModel.ts

```diff
@@ -152,25 +154,50 @@ export class Cognite3DModel extends THREE.Object3D implements CogniteModelBase {
    * const visibleNodes = new TreeIndexNodeCollection(someTreeIndices);
    * model.assignStyledNodeCollection(visibleSet, { rendererGhosted: false });
    * ```
+   * @throws Error if node collection already has been assigned to the model.
    */
   assignStyledNodeCollection(nodeCollection: NodeCollectionBase, appearance: NodeAppearance) {
     trackCadModelStyled(nodeCollection.classToken, appearance);
-    this._styledNodeCollections.push({ nodes: nodeCollection, appearance });
+
+    const index = this._styledNodeCollections.findIndex(x => x.nodeCollection === nodeCollection);
+    if (index !== -1) {
+      throw new Error(
+        'Node collection as already been assigned, use updateStyledNodeCollection() to update the appearance'
+      );
+    }
+
+    this._styledNodeCollections.push({ nodeCollection: nodeCollection, appearance });
     this.cadNode.nodeAppearanceProvider.assignStyledNodeCollection(nodeCollection, appearance);
   }
 
+  /**
+   * Updates styled node collections with a new appearance.
+   * @param nodeCollection    A node collection previously assigned using {@link assignStyledNodeCollection}.
+   * @param newAppearance     New appearance for the nodes in the collection.
+   * @throws Error if node collection hasn't previously been assigned using {@link assignStyledNodeCollection}.
+   */
+  updateStyledNodeCollection(nodeCollection: NodeCollectionBase, newAppearance: NodeAppearance): void {
+    const index = this._styledNodeCollections.findIndex(x => x.nodeCollection === nodeCollection);
+    if (index === -1) {
+      throw new Error('The node collection provide has not been assigned previously');
+    }
+
+    this._styledNodeCollections[index].appearance = newAppearance;
+    this.cadNode.nodeAppearanceProvider.assignStyledNodeCollection(nodeCollection, newAppearance);
+  }
+
   /**
    * Removes styling for previously added styled collection, resetting the style to the default (or
    * the style imposed by other styled collections).
    * @param nodeCollection   Node collection previously added using {@link assignStyledNodeCollection}.
```

Comment:
Should we have a `@throws` clause in this comment as well?

Reply:
Fixed